TimesSquare/TSQCalendarDayButton.m

```diff
@@ -8,31 +8,223 @@
+    self.subtitleSymbolLabel.hidden = !([self.subtitleSymbolLabel.text length] > 0);
+    self.iconImageView.hidden = (self.iconImageView.image == nil);
+
+    CGFloat iconWidth = self.iconImageView.image.size.width;
+    CGFloat iconHeight = self.iconImageView.image.size.height;
+
+    if (! self.subtitleLabel.hidden)
+    {
+        CGSize maxSubtitleSize = [self maxSubtitleSize];
+        CGSize sizeThatFits = [self.subtitleLabel sizeThatFits:maxSubtitleSize];
+
+        CGFloat subtitleWidth = fminf(sizeThatFits.width, maxSubtitleSize.width);
+        CGFloat subtitleHeight = fminf(sizeThatFits.height, maxSubtitleSize.height);
+        CGFloat originX = midX - subtitleWidth/2.0f;
+        CGFloat originY = subtitleCenterY - subtitleHeight/2.0f;
+
+        self.subtitleLabel.frame = CGRectMake(originX,
+                                              originY,
+                                              subtitleWidth,
+                                              subtitleHeight);
+    }
+
+    if (! self.subtitleSymbolLabel.hidden)
+    {
+        CGSize maxSymbolSize = [self maxSubtitleSymboleSize];
+        CGSize sizeThatFits = [self.subtitleSymbolLabel sizeThatFits:maxSymbolSize];
+
+        CGFloat symbolWidth = fminf(sizeThatFits.width, maxSymbolSize.width);
+        CGFloat symbolHeight = fminf(sizeThatFits.height, maxSymbolSize.height);
+        CGFloat originX = CGRectGetMaxX(self.bounds) - TSQCalendarRowCellSubtitleBuffer;
+        CGFloat originY = subtitleCenterY - symbolHeight/2.0f;
+
+        if (! self.subtitleLabel.hidden)
+        {
+            // when subtitle is showing, shift symbol to right of subtitle
+            CGFloat symbolBuffer = (TSQCalendarRowCellSubtitleBuffer - symbolWidth)/2.0f;
+            originX = CGRectGetMaxX(self.subtitleLabel.frame) + symbolBuffer;
+        }
+
+        self.subtitleSymbolLabel.frame = CGRectMake(originX,
+                                                    originY,
+                                                    symbolWidth,
+                                                    symbolHeight);
+    }
+
+    if (! self.iconImageView.hidden)
+    {
+        self.iconImageView.hidden = NO;
+
+        CGFloat midX = CGRectGetMidX(self.bounds);
+
+        CGFloat originX = midX - iconWidth/2.0f;
+        CGFloat originY = subtitleCenterY - iconHeight/2.0f;
+
+        if (! self.subtitleLabel.hidden)
+        {
+            // when subtitle is showing, shift icon to left of subtitle
+            CGFloat iconBuffer = (TSQCalendarRowCellSubtitleBuffer - iconWidth)/2.0f;
+            originX = self.subtitleLabel.frame.origin.x - iconWidth - iconBuffer;
+        }
+
+        self.iconImageView.frame = CGRectMake(originX,
+                                              originY,
+                                              iconWidth,
+                                              iconHeight);
+    }
+}
+
+- (BOOL)isForToday
+{
+    NSDate *today = [NSDate date];
```

Comment:
already spoke offline. it's an existing bug since the framework is agnostic of the 5am boundaries. decided against fixing this since we haven't had any problems.